internal/notifier/pagerduty.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+	eventv1 "github.com/fluxcd/pkg/apis/event/v1beta1"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/jpillora/backoff"
+	"k8s.io/utils/strings/slices"
+)
+
+var (
+	validEndpoints = []string{
+		"events.pagerduty.com",
+		"events.eu.pagerduty.com",
```

Comment:
I'm not sure we'll have to validate these here. Whenever PagerDuty decides to change this list, we'll have to catch up and users won't be able to use the new endpoint address until a new Flux release has been published.

Reply:
I wanted to ensure PagerDuty users that are using the EU servers (such as ourselves) were accounted for. I can remove the validation if you think isn't necessary.

Reply:
I have removed the validation.